src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -269,7 +269,11 @@ private BoundPattern BindDiscardPattern(DiscardPatternSyntax node, TypeSymbol in
                 else if (inputType.IsPointerType() && Compilation.LanguageVersion < MessageID.IDS_FeatureRecursivePatterns.RequiredVersion())
                 {
                     // before C# 8 we did not permit `pointer is null`
-                    diagnostics.Add(ErrorCode.ERR_PointerTypeInPatternMatching, patternExpression.Location);
+                    diagnostics.Add(ErrorCode.ERR_LanguageVersionDoesNotSupportPointerTypeInPatternMatching,
```

Comment:
I suspect #43319 was trying to say that we should use the existing `ERR_FeatureNotAvailableInVersion` diagnostics. This allows the code fixer which updates the language version in the project to "just work". 

This works by adding a member to the [`MessageID` enum](https://github.com/dotnet/roslyn/blob/2cd9c8c97e9e894bd528770fa492df5e35c479f4/src/Compilers/CSharp/Portable/Errors/MessageID.cs#L13), adding a string resource to CSharpResources.resx, then using the [CheckFeatureAvailability](https://github.com/dotnet/roslyn/blob/d1532d6ed1564eb0da2690783f53b588eb6fee42/src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs#L2390) API here.

Perhaps it would make sense to give the new MessageID entry a name like `IDS_FeatureNullPointerConstantPattern`. /cc @333fred to chime in.

Reply:
Thanks for the pointer in the right direction, will have those changes in soon.